src/hooks/tls.ts

```diff
@@ -0,0 +1,48 @@
+import { Options } from 'got-cjs';
+// @ts-expect-error Missing types
+import { getBrowser, getUserAgent } from 'header-generator/src/utils';
+import { ciphers, sigalgs, minVersion, maxVersion } from '../scraping-defaults';
+
+type Browser = 'chrome' | 'firefox' | 'safari' | undefined;
+
+export function tlsHook(options: Options): void {
+    const { https } = options;
+
+    if (https.ciphers || https.signatureAlgorithms || https.minVersion || https.maxVersion) {
+        return;
+    }
+
+    const browser: Browser = getBrowser(getUserAgent(options.headers));
```

Comment:
See comments below, but I think this file suffers a lot from having two sources of truth about which config should be used. One is in this file and other is in scraping defaults. I think we should only have this in one place and scraping defaults seem like the better place, with only simple logic in this file.

I have not given it a lot of thought, but I would probably add safari to the scraping defaults directly like:

```js
ciphers.safari = ciphers.chrome
```

And I would add a comment why we're doing this (Are they the same? Do we not have ciphers for Safari?).

Reply:
> Do we not have ciphers for Safari?

I'll see if we can use https://github.com/cloudflare/mitmengine/tree/master/reference_fingerprints